jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -193,38 +194,15 @@ public RetainableByteBuffer acquire(int size, boolean direct)
         RetainedBucket bucket = bucketFor(size, direct);
         if (bucket == null)
             return newRetainableByteBuffer(size, direct, this::removed);
-        Pool.Entry<RetainableByteBuffer> entry = bucket.getPool().acquire();
 
-        RetainableByteBuffer buffer;
+        Pool.Entry<RetainableByteBuffer> entry = bucket.getPool().acquire();
         if (entry == null)
-        {
-            Pool.Entry<RetainableByteBuffer> reservedEntry = bucket.getPool().reserve();
-            if (reservedEntry != null)
-            {
-                buffer = newRetainableByteBuffer(bucket._capacity, direct, retainedBuffer ->
-                {
-                    BufferUtil.reset(retainedBuffer.getByteBuffer());
-                    if (!reservedEntry.release())
-                        reservedEntry.remove();
-                });
-
-                // A reserved entry may become old and be evicted before it is enabled.
-                if (reservedEntry.enable(buffer, true))
-                {
-                    if (direct)
-                        _currentDirectMemory.addAndGet(buffer.capacity());
-                    else
-                        _currentHeapMemory.addAndGet(buffer.capacity());
-                    releaseExcessMemory(direct);
-                    return buffer;
-                }
-            }
-            return newRetainableByteBuffer(size, direct, this::removed);
-        }
+            return newRetainableByteBuffer(bucket._capacity, direct, this::reserve);
 
-        buffer = entry.getPooled();
-        ((Buffer)buffer).acquire();
-        return buffer;
+        RetainableByteBuffer pooled = entry.getPooled();
+        if (pooled instanceof Buffer buffer)
```

Comment:
Pooled entries are always going to be instances of `Buffer`, aren't they? It looks like the bucket's pool template type isn't right.

Reply:
We have one protected method that leaks the buffer pool, so we'd have to make Buffer public